src/main/java/com/feedzai/commons/sql/abstraction/dml/result/ResultIterator.java

```diff
@@ -56,20 +57,31 @@
     /**
      * The list of columns names for the given query.
      */
-    private final List<String> columnNames;
+    private final List<String> columnNames = new ArrayList<>();
```

Comment:
We don't need random access, so this could be a `LinkedList`.

Reply:
Yes we do; at least at line 239 we are using `columnNames.get(i)`.
I'd have to change that, and it would fall a bit outside of the scope of the issue being addressed